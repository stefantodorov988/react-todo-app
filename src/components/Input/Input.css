.input,
.input:is(:focus, :hover, :active) {
  border: none;
  outline: none;
}

.input::placeholder {
  color: rgb(199, 199, 199);
}

.input {
  font-weight: 500;
  font-size: 1em;
  caret-color: rgb(72, 128, 255);
  width: 100%;
}