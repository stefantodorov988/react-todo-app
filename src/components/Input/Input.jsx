import "./Input.css";

export const Input = ({ handleChange, handleKey, value }) => {
  return (
    <>
      <input
        type="text"
        onChange={handleChange}
        onKeyDown={handleKey}
        value={value}
        className="input"
        placeholder="Please enter text"
      />
    </>
  );
};