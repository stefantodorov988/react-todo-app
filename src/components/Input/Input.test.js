import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Input } from "./Input";

test("renders input field with placeholder", () => {
  const placeholder = "Please enter text";
  const { getByPlaceholderText } = render(<Input placeholder={placeholder} />);

  const input = getByPlaceholderText(placeholder);
  expect(input).toBeInTheDocument();
});

test("calls handleChange when input value changes", () => {
  const handleChange = jest.fn();
  const { getByPlaceholderText } = render(
    <Input handleChange={handleChange} />
  );

  const input = getByPlaceholderText("Please enter text");
  fireEvent.change(input, { target: { value: "New Value" } });
  expect(handleChange).toHaveBeenCalledTimes(1);
});

test("calls handleKey when a key is pressed", () => {
  const handleKey = jest.fn();
  const { getByPlaceholderText } = render(<Input handleKey={handleKey} />);

  const input = getByPlaceholderText("Please enter text");
  fireEvent.keyDown(input, { key: "Enter", code: "Enter" });
  expect(handleKey).toHaveBeenCalledTimes(1);
});

test("renders input field with provided value", () => {
  const value = "Initial Value";
  const { getByPlaceholderText } = render(<Input value={value} />);

  const input = getByPlaceholderText("Please enter text");
  expect(input.value).toBe(value);
});

test("renders Input component correctly", () => {
  const { container } = render(<Input />);
  expect(container.firstChild).toMatchSnapshot();
});
