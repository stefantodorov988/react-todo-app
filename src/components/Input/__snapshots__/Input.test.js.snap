// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input component correctly 1`] = `
<input
  class="input"
  placeholder="Please enter text"
  type="text"
  value=""
/>
`;
